controllers/upgrade_handlers.go

```diff
@@ -21,36 +21,101 @@ import (
 	"fmt"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// log and update status to in-progress
+	r.Log.Info("Starting handleUpgrade")
+	utils.SetStatusCondition(&ibu.Status.Conditions,
+		utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+		utils.ConditionReasons.InProgress,
+		metav1.ConditionTrue,
+		"Upgrade in-progress",
+		ibu.Generation)
+
+	// Pre-pivot steps
+	// backup with OADP
+	r.Log.Info("Starting backup")
+	backupRequeue, backupStatus, err := r.BackupRestore.ReconcileBackup(ctx, ibu.Spec.OADPContent)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("error msg from backup: %s, will requeue after: %s", backupStatus.Message, backupRequeue.RequeueAfter))
+		return backupRequeue, err
+	}
+	switch backupStatus.Status {
+	case backuprestore.BackupCompleted:
+		// only proceed when BackupCompleted
+		r.Log.Info(fmt.Sprintf("Backup completed with msg %s", backupStatus.Message))
+	default:
+		// backup in progress
+		utils.SetStatusCondition(&ibu.Status.Conditions,
+			utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+			utils.ConditionReasons.InProgress,
+			metav1.ConditionFalse,
+			backupStatus.Message,
+			ibu.Generation)
+		return backupRequeue, nil
+	}
 
 	utils.ExecuteChrootCmd(utils.Host, "mount /sysroot -o remount,rw")
 	stateRootRepo := fmt.Sprintf("/host/ostree/deploy/rhcos_%s/var", ibu.Spec.SeedImageRef.Version)
 
-	// TODO: Pre-pivot steps
-	//
-	// TODO: Call r.BackupRestore.ExportRestoresToDir to write restore CRs to the new stateroot.
-	//       If there is invalid type of error returned, we should update the ibu status and
-	//       requeue with short interval.
-	//
-	err := r.ExtraManifest.ExportExtraManifestToDir(ctx, ibu.Spec.ExtraManifests, stateRootRepo)
+	r.Log.Info("Writing extra-manifests into new stateroot")
+	if err = r.ExtraManifest.ExportExtraManifestToDir(ctx, ibu.Spec.ExtraManifests, stateRootRepo); err != nil {
```

Comment:
As we have done the backups previously, could we then export the restore CRs and OADP configurations prior to exporting extramanifests

Reply:
Updated